processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -366,14 +390,24 @@ public void close()
   // here its made thread safe to support the special case of groupBy where the multiple threads can add concurrently to the IncrementalIndex.
   static class ObjectCachingColumnSelectorFactory implements ColumnSelectorFactory
```

Comment:
Could you please also change code of makeXxxColumnSelector methods to use `computeIfAbsent()` instead of `putIfAbsent()`?